### PR TITLE
refactor(asyncTestFn): refactor async test wrapping to show more info

### PR DESCRIPTION
**Pull request by juliemr**
Test wrapping for Jasmine 2 now more closely follows the test wrapping
for Mocha at https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/testing/index.js

This also adds more information to the task names in the control flow,
for easier debugging.

### CODE REVIEW

**Comment by juliemr** [comment]
This is causing some failures when testing against Protractor, not ready for merging yet.

**Comment by juliemr** [comment]
OK! Confirmed that this works with the Protractor suite of tests. @hankduan could you take a look?

File: spec/adapterSpec.js [hunk 1]
```diff
@@ -16,6 +16,21 @@ describe('webdriverJS Jasmine adapter plain', function() {
   });
 });
 
+describe('context', function() {
+  beforeEach(function() {
+    this.foo = 0;
+  });
+
+  it("can use the `this` to share state", function() {
```
Comment (hankduan) [inline spec/adapterSpec.js]: I think linter will complain about double quote

**Comment by juliemr** [comment]
Output of the control flow schedule:

Before:
```
.ControlFlow::1
| Frame::5
| | Frame::10
| | | Frame::12
| | | | (pending) Task::40<Run it in control flow>
| | | | | (active) Frame::48
| | | | | | Task::50<getValueA>
| | | | | | Task::52<getOtherValueA>
| | | | | | Task::54<Expect toEqual>
```

After:
```
.ControlFlow::1
| Frame::4
| | Frame::36
| | | Frame::6
| | | | (pending) Task::38<Run it("should pass async tests") in control flow>
| | | | | (active) Frame::41
| | | | | | Task::44<getValueA>
| | | | | | Task::46<getOtherValueA>
| | | | | | Task::48<Expect toEqual>
```

**Comment by hankduan** [comment]
LGTM

**Pull request merged**